### PR TITLE
Used flex instead of floats for header layout

### LINKED ISSUE #1435: When the "hamburger" menu is open, pressing the tab key for the first time should bring the focus to the first item in the menu list
When the menu is open, the first time we press the tab key, it should bring the focus to the first menu item on the list. Currently, it jumps to the light/dark theme button.

### PR DESCRIPTION
**Pull request by bmispelon**
Fixed #1435

### CODE REVIEW

File: djangoproject/templates/includes/header.html [hunk 1]
```diff
@@ -1,14 +1,15 @@
 <div role="banner" id="top">
-  <div class="container">
+  <div class="container container--flex--wrap--mobile">
     <a class="logo" href="{% url 'homepage' %}">Django</a>
     <p class="meta">The web framework for perfectionists with deadlines.</p>
+    <div class="spacer"></div>
```
Comment (bmispelon) [inline djangoproject/templates/includes/header.html:5]: I don't like having this in the HTML but I couldn't figure out how to achieve the same result with CSS only. Suggestions welcome!

Comment (sabderemane) [reply djangoproject/templates/includes/header.html:5]: Did you try to use `justify-content: space-around` or `justify-content: space-between`? (I didn't try, might not work but I can have a look later)

**Review by sabderemane — COMMENTED** [review]
Looks nice! ✨

File: djangoproject/templates/includes/header.html (hunk 1 above)
Comment (matthiask) [reply djangoproject/templates/includes/header.html:5]: Another thing which might work is `margin-left: auto;` on `.mobile-toggle`. Margins sometimes work surprisingly well inside flex containers.

Comment (bmispelon) [reply djangoproject/templates/includes/header.html]: Finally got around to trying the `margin-left` approach and it's working perfectly. Thanks for the tip! 👍🏻

Comment (matthiask) [reply djangoproject/templates/includes/header.html]: Nice!

**Comment by bmispelon** [comment]
Thanks everyone! ✨

**Pull request merged**